File: js/management-api-v2/cm_api_v2_post_snippet.js
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: '<YOUR_ENVIRONMENT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

const response = await client.addContentTypeSnippet()
  .withData(builder => {
    return {
      name: "metadata",
      codename: "my_metadata",
      external_id: "snippet-item-123",
      elements: [
        builder.textElement({
          name: "Meta title",
          codename: "my_metadata__meta_title",
          guidelines: "Length: 30–60 characters",
          type: ElementModels.ElementType.text,
          external_id: "meta_title"
        }),
        builder.textElement({
          name: "Meta description",
          codename: "my_metadata__meta_description",
          guidelines: "Length: 70-150 characters",
          type: ElementModels.ElementType.text,
          external_id: "meta_title"
        })
      ]
    };
  })

  .toPromise();

File: net/management-api-v2/cm_api_v2_post_snippet.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_ENVIRONMENT_ID>"
});

var response = await client.CreateContentTypeSnippetAsync(new ContentTypeSnippetCreateModel
{
    Name = "metadata",
    Codename = "my_metadata",
    ExternalId = "snippet-item-123",
    Elements = new ElementMetadataBase[]
    {
        new TextElementMetadataModel
        {
            Name = "Meta title",
            Codename = "my_metadata__meta_title",
            Guidelines = "Length: 30-60 characters",
            ExternalId = "meta_title",
        },
        new TextElementMetadataModel
        {
            Name = "Meta description",
            Codename = "my_metadata__meta_description",
            Guidelines = "Length: 70-11500 characters",
            ExternalId = "meta_description",
        }
    }
});
